Gems/Terrain/Code/Source/TerrainRenderer/Passes/TerrainClipmapComputePass.h

```diff
@@ -12,19 +12,78 @@
 #include <Atom/RPI.Public/Pass/ComputePass.h>
 #include <Atom/RPI.Public/Pass/ParentPass.h>
 #include <Atom/RPI.Public/Pass/Pass.h>
+#include <Atom/RPI.Public/Shader/ShaderReloadNotificationBus.h>
 #include <Atom/RPI.Reflect/Pass/ComputePassData.h>
 #include <Atom/RPI.Reflect/Pass/PassDescriptor.h>
 #include <TerrainRenderer/TerrainClipmapManager.h>
 
+#include <Atom/RPI.Reflect/Pass/RenderPassData.h>
+
 namespace Terrain
 {
+    struct TerrainClipmapGenerationPassData : public AZ::RPI::RenderPassData
+    {
+        AZ_RTTI(TerrainClipmapGenerationPassData, "{07C90E11-6607-4BD2-B041-96CEF46F8C55}", AZ::RPI::RenderPassData);
+        AZ_CLASS_ALLOCATOR(TerrainClipmapGenerationPassData, AZ::SystemAllocator, 0);
+
+        TerrainClipmapGenerationPassData() = default;
+        virtual ~TerrainClipmapGenerationPassData() = default;
+
+        static void Reflect(AZ::ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+            {
+                serializeContext->Class<TerrainClipmapGenerationPassData, AZ::RPI::RenderPassData>()->Version(1)->Field(
+                    "ShaderAsset", &TerrainClipmapGenerationPassData::m_shaderReference);
+            }
+        }
+
+        AZ::RPI::AssetReference m_shaderReference;
+    };
+
+    class TerrainClipmapGenerationPass
```

Comment:
Class comment please - a bit about what this pass does so readers new to the system have some context.